Data_Structures_Answers.md

```diff
@@ -2,17 +2,31 @@ Add your answers to the questions below.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
 
+its just straight up number maniputlation and so its linear O(1)
+
 2. What is the space complexity of your ring buffer's `append` function?
 
+still linear and with variable exhange but always the final variable solution, so O(n)
+
 3. What is the runtime complexity of your ring buffer's `get` method?
 
+O(n) either way!
 4. What is the space complexity of your ring buffer's `get` method?
 
+doesnt change at all, very flat like a O(n)
 
 5. What is the runtime complexity of the provided code in `names.py`?
 
+ran like the robot sort and so there are two lists looping, so O(n^2)
+
 6. What is the space complexity of the provided code in `names.py`?
 
+became a straight line to the end so one answer fluctiating to become itself at the end, O(n)
+
 7. What is the runtime complexity of your optimized code in `names.py`?
 
+still O(n)
+
 8. What is the space complexity of your optimized code in `names.py`?
```

Comment:
I think your code was some of the most clever in the group, Laryna! Make sure to keep the fire going for this week in graphs!